### ISSUE TITLE
tex-mml-svg.js:1 Uncaught Error: Unknown node type “body” while creating MathJax SVG

### ISSUE THREAD

**Issue by imabot2** [mathjax/MathJax#2389]
I try to create an SVG equation with MathJax according to http://docs.mathjax.org/en/latest/web/typeset.html#converting-a-math-string-to-other-formats 

It works fine for inline equations:

    \( e=mc^2 \)

But an error occurred while creating the SVG in JavaScript :

    MathJax.mathml2svg("x_2 = y^3")

Here is the error:

    tex-mml-svg.js:1 Uncaught Error: Unknown node type "body"
        at o.error (tex-mml-svg.js:1)
        at o.makeNode (tex-mml-svg.js:1)
        at o.compile (tex-mml-svg.js:1)
        at c.compile (tex-mml-svg.js:1)
        at T.e.compile (tex-mml-svg.js:1)
        at Object.renderMath (tex-mml-svg.js:1)
        at d.renderConvert (tex-mml-svg.js:1)
        at T.e.convert (tex-mml-svg.js:1)
        at t.Z.convert (tex-mml-svg.js:1)
        at Object.v.MathJax.<computed> [as mathml2svg] (tex-mml-svg.js:1)

Not sure if this is a bug or if I missed something. Here a JSFiddle illustrating the problem: 

* https://jsfiddle.net/Imabot/xkud6o73/1/
* https://jsfiddle.net/Imabot/xkud6o73/5/

**Comment by dpvc** [comment]
This is a duplicate of issue #2330.  Please see that issue for additional details.

**Comment by imabot2** [comment]
@dpvc are you sure this is the same issue? I do not even understand the code in the question! Did you have a look at my JSFiddles ?

**Comment by dpvc** [comment]
Yes, I'm sure.  There are several different ways of calling MathJax, and it turns out that you are using a different one than in the question I cited, so that is why the code looks strange to you.

The issue is resolved in a pull request that will be part of v3.0.2.  The work-around for the problem (as described in [my comment](https://github.com/mathjax/MathJax/issues/2330#issuecomment-586311389) in the cited issue) is to set the `forceReparse` option for the MathML input jax.  In your case, you would need to use

``` html
<script>
MathJax = {
  mml: {forceReparse: true}
}
</script>
```

somewhere *before* loading MathJax itself.

This doesn't work directly in your jsFiddle because you are loading the `tex-mml-svg.js` component as a jsFiddle resource (so it will be loaded before the script above runs, and the script will then overwrite the MathJax variable that MathJax set up).  Your later call to cdnjs to load MathJax again has the `async` attribute, so it will not block, and the rest of the page is processed while MathJax is being loaded (a second time).  With the `MathJax = {...}` in place, which replaces the MathJax loaded as a resource before your page runs, the script that calls `MathJax.mathml2svg()` will fail because the second loading of MathJax hasn't taken place yet.  (The only reason your original jsFiddle ran was that MathJax had been preloaded as a resource, not because of the `<script>` tag that loads it).

So to get it to work in your jsFiddle, you need to remove that resource first.  That is, you need to remove the `tex-mml-svg.js` resource and let the script tag load MathJax so that its configuration will work properly.  But because the script that loads MathJax is `async`, you will still have the problem that `MathJax.math2svg()` is not yet available when your body script runs.  You could remove the `async` attribute, but because you are calling the `latest` script, that will look up the latest version and load that, but that loading is asynchronous, so you will still run into the same issue with `MathJax.math2svg()` not being defined yet.

So you will need to use the MathJax startup synchronization mechanisms in order to make sure that your code waits for MathJax to be ready before you call it.  That is done through the `startup.pageReady()` function.  Here is one version that does that:

``` html
<script>
  MathJax = {
    mml: {forceReparse: true},
    startup: {
      pageReady () {
        MathJax.startup.defaultPageReady().then(() => {
          try {
            console.log (MathJax.mathml2svg("x_2 = y^3"))
          } catch(e) {
            console.log(e);
          }
        });
      }
    }
  };
</script>
```

Unfortunately, there are a couple of other problems in your jsFiddle.  For one, you need to use `lates.js`, not just `latest`.  Also, it looks like cdnjs never picked up version 3.0.0, so you need to use 3.0.1 or the file won't be found.

Fixing all of those issues, together with the `forceReparse` option, will get you a version that runs properly, but there is still one problem:  you are not passing `MathJax.mathml2svg()` a MathML expression, which would look like `"<math><msub><mi>x</mi><mn>2</mn></msub>...</math>"`.  Perhaps your input is actually LaTeX?  In that case, use `MathJax.tex2svg()` instead.  As it stands, you will get an error from `MathJax.mathml2svg("x^2 = y^3")` indicating that `MathML must be formed by a <math> element not <#text>`.

A fully working example would be

``` html
<html>
<head>
  <script>
  MathJax = {
    mml: {forceReparse: true},
    startup: {
      pageReady () {
        MathJax.startup.defaultPageReady().then(() => {
          try {
            console.log (MathJax.tex2svg("x_2 = y^3"))
          } catch(e) {
            console.log(e);
          }
        });
      }
    }
  }
  </script>
  <script type="text/javascript" id="MathJax-script" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/3.0.1/es5/latest.js?tex-mml-svg.js">
  </script>
</head>
<body>
  Working: \( x^2\)
</body>
</html>
```

So yes, I'm sure that is the issue, and yes, I did look at your jsFiddle.

**Comment by imabot2** [comment]
You are awesome, thank you so much !
